packages/theme/src/theme.ts

```diff
@@ -133,8 +158,13 @@ export class Theme extends HTMLElement {
     }
 
     protected connectedCallback(): void {
+        /* istanbul ignore if */
+        if (!this.hasAdoptedStyles) {
```

Comment:
Before we were doing this in the constructor, but it means that it might run before the attributes are set. This should help reduce some edge case extra renders.